crates/nu-cli/src/completions/file_completions.rs

```diff
@@ -105,6 +105,19 @@ pub fn partial_from(input: &str) -> (String, String) {
     // rsplit_once removes the separator
     base.push(SEP);
 
+    let num_leading_dots = rest.bytes().take_while(|c| *c == b'.').count();
+
+    let (leading_dots, following_leading_dots) = rest.split_at(num_leading_dots);
+
+    // only push dots from rest into base if we believe input is not a prefix to a single dot hidden folder
+    // TODO: Also deal with legal but rare files or folders that contain multiple leading dots.
+    if num_leading_dots > 1 {
+        base.push_str(leading_dots);
+        base.push(SEP);
+
+        return (base.to_string(), following_leading_dots.to_string());
+    }
+
```

Comment:
Not sure if that logic might have to defer some of that decision to the following file and directoy completers